cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/mds/AbstractMDSUpgrader.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.data2.datafabric.dataset.service.mds;
```

Comment:
why is it in this package?
I don't think the upgrade is a function of the MDS. 
It is operating on the MDS, but it is a function of either the upgrade tool or the dataset service. But it seems that even DatasetService does not understand what a dataset does, it just manages its code and meta data. So I think this belongs into the upgrade tool.